File: Backup Assistant/Core/IBackupStarter.vb
Imports System.Collections.ObjectModel

Namespace Core

    Friend Interface IBackupStarter

        ReadOnly Property SourcePath As String

        ReadOnly Property DestinationPath As String

        ReadOnly Property Filters As ReadOnlyCollection(Of String)

        Sub PreProcess()

        Sub PostProcess()

        Sub ReportProgress(progress As Integer)

        Sub ReportStatus(status As String)

        Sub AddToLogEntry(message As String)

    End Interface

End Namespace

File: src/Core/IBackupStarter.cs
using System.Collections.Generic;


namespace BackupAssistant.Core
{
    public interface IBackupStarter
    {
        string SourcePath { get; }

        string DestinationPath { get; }

        IList<string> Filters { get; }

        void PreProcess();

        void PostProcess();

        void ReportProgress(int progress);

        void ReportStatus(string status);

        void AddToLogEntry(string message);
    }
}
